instrumentation/opentelemetry-instrumentation-urllib3/src/opentelemetry/instrumentation/urllib3/__init__.py

```diff
@@ -47,14 +47,19 @@ def strip_query_params(url: str) -> str:
 
 .. code:: python
 
-    # `request` is an instance of urllib3.connectionpool.HTTPConnectionPool
-    def request_hook(span, request):
-        pass
-
-    # `request` is an instance of urllib3.connectionpool.HTTPConnectionPool
-    # `response` is an instance of urllib3.response.HTTPResponse
-    def response_hook(span, request, response):
-        pass
+    def request_hook(
+        span: Span,
+        pool: urllib3.connectionpool.HTTPConnectionPool,
+        request_info: RequestInfo,
```

Comment:
Nit: Anyway to include a link to the definition or include the fields necessary so it's easier for users to create one.

Reply:
Could you elaborate on why users would need to create one? In the current form, it's used to pass the info from the instrumentation down to the hook, the user code wouldn't need to create one in the first place

Reply:
I think @lzchen is talking about cases where users just copy and paste the example from the documentation; see https://opentelemetry-python-contrib.readthedocs.io/en/latest/instrumentation/urllib3/urllib3.html. If we are providing the example with the type would be nice to have a reference to it, or just omit it from the example

Reply:
I believe, all the public members should be automagically rendered in the API section, right? And then, I thought, having the type hint would be beneficial. I'm not sure if it's even possible to make this type hint inside the snippet a proper link

Reply:
> the user code wouldn't need to create one in the first place

Apologies, instead of "create" I meant "they need to know what a `RequestInfo` consists of" for them to use it properly in their hook. As @emdneto mentioned, if a user were to just look at the doc example and copy and paste it, it would be convenient to immediately have the class definition so as to know how to use it. A simple comment would suffice. However, not a blocker.